test/unit/hostgroup_test.rb

```diff
@@ -357,14 +357,41 @@ def setup_user(operation)
     assert_equal "#{hostgroup.id}-a-b",  hostgroup.to_param
   end
 
-  test "clone should clone config groups as well" do
-    group = FactoryGirl.create(:hostgroup, :name => 'a')
-    config_group = ConfigGroup.create!(:name => 'Blah')
-    group.config_groups << config_group
-    group.save
-
-    cloned = group.clone("new_name")
-    assert cloned.config_groups.include?(config_group)
+  context "#clone" do
+    let(:group) {FactoryGirl.create(:hostgroup, :name => 'a')}
+
+    test "clone should clone config groups as well" do
+      config_group = ConfigGroup.create!(:name => 'Blah')
+      group.config_groups << config_group
+      group.save
```

Comment:
Then add more tests if you want to ensure that behavior (although I think it's much better as it is now, you're testing clone, not validations), currently tests will pass if you remove all the `.save` calls in this context group (except for the lookup_value).